// uikit/src/js/components/slideshow.js
import { css } from 'uikit-util';
import Class from 'uikit/src/js/mixin/class';
import SliderParallax from 'uikit/src/js/mixin/slider-parallax';
import SliderReactive from 'uikit/src/js/mixin/slider-reactive';
import Slideshow from 'uikit/src/js/mixin/slideshow';
import SliderPreload from 'uikit/src/js/components/internal/slider-preload';
import Animations from 'uikit/src/js/components/internal/slideshow-animations';

export default {
    mixins: [Class, Slideshow, SliderReactive, SliderParallax, SliderPreload],

    props: {
        ratio: String,
        minHeight: String,
        maxHeight: String,
    },

    data: {
        ratio: '16:9',
        minHeight: undefined,
        maxHeight: undefined,
        selList: '.uk-slideshow-items',
        attrItem: 'uk-slideshow-item',
        selNav: '.uk-slideshow-nav',
        Animations,
    },

    watch: {
        list(list) {
            css(list, {
                aspectRatio: this.ratio ? this.ratio.replace(':', '/') : undefined,
                minHeight: this.minHeight,
                maxHeight: this.maxHeight,
                width: '100%',
            });
        },
    },

    methods: {
        getAdjacentSlides() {
            return [1, -1].map((i) => this.slides[this.getIndex(this.index + i)]);
        },
    },
};
